src/fs/filesystem.ts

```diff
@@ -157,11 +187,20 @@ export class FileSystem {
     ]
   }
 
+  async shareWith(did: string): Promise<CID> {
+    const ks = await keystore.get()
+    const pubkey = didUtil.didToPubKey(did)
+    const encryptedRootKey = await ks.encrypt(this.key, pubkey)
+    this.keyTree.add(did, encryptedRootKey)
+    return this.sync()
+  }
+
   async sync(): Promise<CID> {
     this.root = await asyncWaterfall(this.root, [
       (t: Tree) => t.addChild('public', this.publicTree),
       (t: Tree) => t.addChild('pretty', this.prettyTree),
-      (t: Tree) => t.addChild('private', this.privateTree)
+      (t: Tree) => t.addChild('private', this.privateTree),
```

Comment:
the more trees we add the more needlessly expensive `sync`s get...

Mostly a note to myself: I'd like to rewrite this logic soon

Reply:
Okay changed this around to update the root node every time a child is updated so that `sync` only has to `put` the root